datasources/beeper/beeper.go

```diff
@@ -0,0 +1,982 @@
+	offset := 0
+	batchSize := 100
+	totalProcessed := 0
+
+	for {
+		// Build a query for threads with pagination
+		query := fmt.Sprintf("SELECT * FROM threads LIMIT %d OFFSET %d", batchSize, offset)
+		params.Log.Info("Executing threads query batch",
+			zap.String("query", query),
+			zap.Int("offset", offset),
+			zap.Int("processed", totalProcessed),
+			zap.Int("total", totalThreads))
+
+		// Execute the query
+		rows, err := db.QueryContext(ctx, query)
+		if err != nil {
+			return fmt.Errorf("executing threads query: %w", err)
+		}
+
+		// Process the rows
+		count := 0
+		for rows.Next() {
+			select {
+			case <-ctx.Done():
+				rows.Close()
+				return ctx.Err()
+			default:
+				// Get column names
+				columns, err := rows.Columns()
+				if err != nil {
+					rows.Close()
+					return fmt.Errorf("getting column names: %w", err)
+				}
+
+				// Create a slice of interface{} to hold the values
+				values := make([]interface{}, len(columns))
+				valuePtrs := make([]interface{}, len(columns))
+
+				// Create pointers to the values
+				for i := range values {
+					valuePtrs[i] = &values[i]
+				}
+
+				// Scan the row
+				if err := rows.Scan(valuePtrs...); err != nil {
+					rows.Close()
+					return fmt.Errorf("scanning row: %w", err)
+				}
+
+				// Find important column indices
+				threadIDIdx := findColumnIndex(columns, "threadID")
+				accountIDIdx := findColumnIndex(columns, "accountID")
+				threadIdx := findColumnIndex(columns, "thread")
+				timestampIdx := findColumnIndex(columns, "timestamp")
+
+				// Extract the values
+				var threadID, accountID, threadJSON string
+				var timestamp time.Time
+
+				if threadIDIdx >= 0 && threadIDIdx < len(values) {
+					threadID = toString(values[threadIDIdx])
+				}
+
+				if accountIDIdx >= 0 && accountIDIdx < len(values) {
+					accountID = toString(values[accountIDIdx])
+				}
+
+				if threadIdx >= 0 && threadIdx < len(values) {
+					threadJSON = toString(values[threadIdx])
+				}
+
+				if timestampIdx >= 0 && timestampIdx < len(values) {
+					timestamp = toTimestamp(values[timestampIdx])
+				} else {
+					timestamp = time.Now()
+				}
+
+				// Try to extract the thread name from JSON
+				var threadName string
+				if strings.Contains(threadJSON, "\"name\":") {
+					// This is a very simple JSON extraction - in a real implementation
+					// you would properly parse the JSON
+					parts := strings.Split(threadJSON, "\"name\":")
+					if len(parts) > 1 {
+						namePart := parts[1]
+						if strings.Contains(namePart, "\"") {
+							namePart = strings.Split(namePart, "\"")[1]
+							threadName = namePart
+						}
+					}
+				}
+
+				if threadName == "" {
+					threadName = "Chat " + threadID
+				}
+
+				// Create entity for the thread/conversation
+				entity := &timeline.Entity{
+					Name: threadName,
+					Attributes: []timeline.Attribute{
+						{
+							Name:     "beeper_thread_id",
+							Value:    threadID,
+							Identity: true,
```

Comment:
Is the beeper thread this person's identity? Or just something they are participating in? If the latter, this shouldn't be an attribute I think.